CRM/Event/Form/SelfSvcTransfer.php

```diff
@@ -288,7 +295,8 @@ public static function checkRegistration($fields, $self, $contact_id, &$errors)
     if (!empty($to_event_id)) {
       foreach ($to_event_id as $id) {
         if ($id == $self->_event_id) {
-          $errors['email'] = $display_name . ts(" is already registered for this event");
+          $error_field = $fields['contact_id'] ? 'contact_id' : 'email';
+          $errors[$error_field] = $display_name . ts(" is already registered for this event");
```

Comment:
I know this wasn't your doing, but this usage of `ts()` [is incorrect](https://docs.civicrm.org/dev/en/latest/translation/). It should be 
```suggestion
          $errors[$error_field] = ts('%1 is already registered for this event', [1 => $display_name]);
```